File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/customviews/progressDialog/CancelableProgressDialog.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.customviews.progressDialog;

import android.app.ProgressDialog;
import android.content.Context;

public class CancelableProgressDialog implements Dialog {

    public ProgressDialog dialog;

    @Override
    public void showDialog(Context context, String msg) {
        dialog = new ProgressDialog(context);
        dialog.setMessage(msg);
        dialog.setCancelable(true);
        dialog.setCanceledOnTouchOutside(true);
        dialog.show();
    }

    @Override
    public void dismiss() {
        dialog.dismiss();
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/presentation/customviews/progressdialog/CancelableProgressDialog.kt
@file:Suppress("DEPRECATION")

package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.presentation.customviews.progressdialog

import android.app.ProgressDialog
import android.content.Context

class CancelableProgressDialog : Dialog {

    private lateinit var dialog: ProgressDialog

    override fun showDialog(context: Context, msg: String) {
        dialog = ProgressDialog(context)
        dialog.setMessage(msg)
        dialog.setCancelable(true)
        dialog.setCanceledOnTouchOutside(true)
        dialog.show()
    }

    override fun dismiss() {
        dialog.dismiss()
    }
}
